modules/addefendBidAdapter.js

```diff
@@ -0,0 +1,79 @@
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+
+const BIDDER_CODE = 'addefend';
+
+export const spec = {
+  code: BIDDER_CODE,
+  hostname: 'https://addefend-platform.com',
+
+  getHostname() {
+    return this.hostname;
+  },
+  isBidRequestValid: function(bid) {
+    return (bid.sizes !== undefined && bid.bidId !== undefined && bid.params !== undefined &&
+              (bid.params.pageId !== undefined && (typeof bid.params.pageId === 'string')) &&
+              (bid.params.placementId !== undefined && (typeof bid.params.placementId === 'string')));
+  },
+  buildRequests: function(validBidRequests, bidderRequest) {
+    let bid = {
+      v: $$PREBID_GLOBAL$$.version,
+      auctionId: false,
+      pageId: false,
+      gdpr_consent: bidderRequest.gdprConsent && bidderRequest.gdprConsent.consentString ? bidderRequest.gdprConsent.consentString : '',
+      referer: bidderRequest.refererInfo.referer,
+      bids: [],
+    };
+
+    for (var i = 0; i < validBidRequests.length; i++) {
+      let vb = validBidRequests[i];
+      let o = vb.params;
+      bid.auctionId = vb.auctionId;
```

Comment:
in this loop does it mean that `bid.blah` will get overridden every time?

Reply:
yes, auctionId, pageId and trafficTypes are supposed to be the same for all the requests.